animation_nodes/nodes/matrix/distribute_matrices.pyx

```diff
@@ -260,9 +265,19 @@ class DistributeMatricesNode(bpy.types.Node, AnimationNode):
         stepSin = sin(angleStep)
 
         for i in range(amount):
-            vector.x = iCos * radius
-            vector.y = iSin * radius
-            vector.z = 0
+            if self.planeAxis == "XY":
+                vector.x = iCos * radius
+                vector.y = iSin * radius
+                vector.z = 0
+            elif self.planeAxis == "YZ":
+                vector.x = 0
+                vector.y = iSin * radius
+                vector.z = iCos * radius
+            else:
+                vector.x = iCos * radius
+                vector.y = 0
+                vector.z = iSin * radius
```

Comment:
Move the condition outside of the loop. This is too much overhead inside the loop.

Reply:
Okay.